### ISSUE TITLE
Production CSS optimization breaks margins behavior

### ISSUE THREAD

**Issue by aspirisen** [vercel/next.js#45229]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.20.0
    Relevant packages:
      next: 13.1.6-canary.0
      eslint-config-next: 13.0.6
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/modest-water-cc23eb?file=%2Fpackage.json&selection=%5B%7B%22endColumn%22%3A25%2C%22endLineNumber%22%3A14%2C%22startColumn%22%3A25%2C%22startLineNumber%22%3A14%7D%5D

### To Reproduce

1. Use `app` dir
2. Create css file 
```
.test {
    --test-a: initial;
    --test-b: initial;
    --test-c: 10px;
    --test-d: initial;

    margin-top: var(--test-a);
    margin-left: var(--test-b);
    margin-bottom: var(--test-c);
    margin-right: var(--test-d);
}
```
4. Use this css in your application
5. Build aplication in production mode and run it

### Describe the Bug

All `margins` props are combined into one `margin` prop. Because of that whole margin property is broken.
I thought that it was `postss` and tried to make it minimal:
```
# postcss.config.json
{
    "plugins": ["postcss-nesting"]
}

```
But that didn't help

### Expected Behavior

There should be a way to disable optimization because it breaks production behavior.

### Which browser are you using? (if relevant)

Chrome 109.0.5414.87

**Comment by SukkaW** [comment]
> All `margins` props are combined into one `margin` prop. Because of that whole margin property is broken. I thought that it was `postcss` and tried to make it minimal:

Would you like to share with me the output of the CSS?

Theoretically, the CSS minifier should have emitted `margin: var(--test-a) var(--test-b) var(--test-c) var(--test-d);`, which would have the same behavior as your source code.

**Comment by aspirisen** [comment]
Yes, it emits `var(--test-a) var(--test-d) var(--test-c) var(--test-b)`
But that doesn't work

Here is the result with original source code where margins are not combined
<img width="578" alt="image" src="https://user-images.githubusercontent.com/3620639/215323532-325d9e82-e378-4ad8-a617-24623788fffa.png">

Here is the optimized result
<img width="614" alt="image" src="https://user-images.githubusercontent.com/3620639/215323548-50e8c939-4b40-42bf-abaa-ec451d77ae67.png">
